bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/handler/ProcessInfoHandler.java

```diff
@@ -0,0 +1,200 @@
+                    break;
+                }
+                case CHANNEL_PROCESS_RESIDENT_MEMORY: {
+                    BigDecimal state = systeminfo.getProcessResidentMemory(pid);
+                    updateState(channelUID, new QuantityType<>(state, SmartHomeUnits.BYTE));
+                    break;
+                }
+                case CHANNEL_PROCESS_VIRTUAL_MEMORY: {
+                    BigDecimal state = systeminfo.getProcessVirtualMemory(pid);
+                    updateState(channelUID, new QuantityType<>(state, SmartHomeUnits.BYTE));
+                    break;
+                }
+                case CHANNEL_PROCESS_USER: {
+                    String state = systeminfo.getProcessUser(pid);
+                    updateState(channelUID, new StringType(state));
+                    break;
+                }
+                case CHANNEL_PROCESS_PATH: {
+                    String state = systeminfo.getProcessPath(pid);
+                    updateState(channelUID, new StringType(state));
+                    break;
+                }
+                case CHANNEL_PROCESS_THREADS: {
+                    BigDecimal state = systeminfo.getProcessThreads(pid);
+                    updateState(channelUID, new DecimalType(state));
+                    break;
+                }
+                default: {
+                    logger.debug("Channel with unknown ID: {}.", channelUID);
+                    break;
+                }
+            }
+        } catch (IllegalArgumentException exception) {
+            if (isLinked(channelUID)) {
+                logger.warn("No information for channel {} with process id {}.", channelUID, pid);
+            }
+            updateState(channelUID, UnDefType.UNDEF);
+        } catch (Exception exception) {
+            String message = exception.getMessage();
+            logger.debug("Unexpected error occurred while getting system information: {}.", message);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing.");
+
+        SystemInfoHandler handler = null;
+        final Thing thing = getThing();
+        final Bridge bridge = getBridge();
+        if (updateConfiguration(thing) && (bridge != null)) {
+            handler = (SystemInfoHandler) bridge.getHandler();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED, "Thing cannot be initialized.");
+        }
+        if (handler != null) {
+            for (Channel channel : thing.getChannels()) {
+                Configuration properties = channel.getConfiguration();
+                handler.changeChannelPriority(channel.getUID(), (String) properties.get(PARAMETER_PRIOIRITY));
+            }
+            logger.debug("Thing is successfully initialized.");
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
The online state is already logged by the framework. So, the log message could be removed.